src/main/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAligner.scala

```diff
@@ -25,21 +25,36 @@
 package com.fulcrumgenomics.alignment
 
 import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.alignment.NeedlemanWunschAligner.Mode.{Global, Glocal}
 import com.fulcrumgenomics.alignment.NeedlemanWunschAligner._
+import enumeratum.EnumEntry
 import htsjdk.samtools.CigarOperator
 
-import scala.collection.mutable
+import scala.collection.{immutable, mutable}
 import scala.math.max
 
 object NeedlemanWunschAligner {
+
+  sealed trait Mode extends EnumEntry
+
+  /** Enum to represent alignment modes supported by the NeedlenameWunschAligner */
+  object Mode extends FgBioEnum[Mode] {
+    /** Alignment mode for global pairwise alignment. */
+    case object Global extends Mode
+    /** Alignment mode for global alignment of query sequence to local region of target sequence. */
+    case object Glocal extends Mode
```

Comment:
I see four possible modes:
1. global (full query to full target)
2. glocal (full query in a subsequence of target)
3. gLeftLocal (prefix of query in a subsequence of target)
4. gRightLocal (suffix of query in a subsequence of target)

I don't think it would be hard to add those four modes in now since glocal already requires the start/end rules, whereas 3-4 only applies the glocal start/end rule respectively.

Reply:
I don't disagree, but I don't (yet) see the utility for 3 & 4, so I'm going to skip for now.